src/python/zquantum/core/interfaces/ansatz.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABC, abstractmethod
+from ..circuit import Circuit
+from .ansatz_utils import ansatz_property
+from ..utils import create_symbols_map
+import copy
+import sympy
+from typing import List, Optional
+from overrides import EnforceOverrides
+import numpy as np
+
+
+class Ansatz(ABC, EnforceOverrides):
+
+    supports_parametrized_circuits = None
+    n_layers = ansatz_property("n_layers")
+
+    def __init__(self, n_layers: int):
+        """
+        Interface for implementing different ansatzes.
+        This class also caches the circuit for given ansatz parameters.
+
+        Args:
+            n_layers: number of layers of the ansatz
+        
+        Attributes:
+            n_layers (int): see Args
+            circuit (zquantum.core.circuit.Circuit): circuit representation of the ansatz.
```

Comment:
I believe this is now `parameterized_circuit`. Docstring should be updated accordingly